src/test/ui/parser/lifetime-in-pattern.stderr

```diff
@@ -2,7 +2,16 @@ error: unexpected lifetime `'a` in pattern
   --> $DIR/lifetime-in-pattern.rs:1:10
    |
 LL | fn test(&'a str) {
-   |          ^^ unexpected lifetime
+   |          ^^
+   |          |
+   |          unexpected lifetime
+   |          help: remove the lifetime
 
-error: aborting due to previous error
+error: expected one of `:`, `@`, or `|`, found `)`
+  --> $DIR/lifetime-in-pattern.rs:1:16
+   |
+LL | fn test(&'a str) {
+   |                ^ expected one of `:`, `@`, or `|`
```

Comment:
This looks like something we should handle better :-/

Reply:
I think the solution here is to make the type ascription optional syntactically. This would allow us to handle this in e.g. resolve or typeck to handle this well without any hacks.